main.py

```diff
@@ -122,138 +140,213 @@ def getArcGISGroupByTitle(arcGISAdmin, title):
+            logger.info('Exception while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+    if output:
+        logger.info('Unexpected output while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, output))
+    usersNotRemoved = results.get('notRemoved')
+    """:type usersNotRemoved: list"""
+    if usersNotRemoved:
+        logger.warning('Warning: Some or all users not removed from ArcGIS group {}: {}'.format(groupNameAndID, usersNotRemoved))
+        
+    return results
+
+
+def removeExistingGroupMembers(groupTitle, group,instructorLog,groupUsers):
+    """Get list of ArgGIS users to remove from group and call method to remove them."""
+    results = ''
+    groupNameAndID = util.formatNameAndID(group)
+    logger.info('Found ArcGIS group: {}'.format(groupNameAndID))
+    instructorLog += 'Updating ArcGIS group: "{}"\n'.format(groupTitle)
+    
+    if not groupUsers:
+        logger.info('Existing ArcGIS group {} does not have users to remove.'.format(groupNameAndID))
+    else:
+        results = removeListOfUsersFromArcGISGroup(group, groupNameAndID, groupUsers)
+        
+    return instructorLog, results
+
+
+def createNewArcGISGroup(arcGIS, groupTags, groupTitle,instructorLog):
+    """Create a new ArgGIS group.  Return group and any creation messages."""
+    logger.info('Creating ArcGIS group: "{}"'.format(groupTitle))
+    instructorLog += 'Creating ArcGIS group: "{}"\n'.format(groupTitle)
+    with util.CaptureStdoutLines() as output:
+        try:
+            arcGISOrgTools = orgtools.orgtools(arcGIS)
+            group = arcGISOrgTools.createGroup(groupTitle, groupTags)
+        except Exception as exception:
+            logger.info('Exception while creating ArcGIS group "{}": {}'.format(groupTitle, exception))
+    if output:
+        logger.info('Unexpected output while creating ArcGIS group "{}": {}'.format(groupTitle, output))
+    return group, instructorLog
+
+
+# Get ArcGIS group with this title (if it exists)
+def lookForExistingArcGISGroup(arcGIS, groupTitle):
+    """Find an ArgGIS group with a matching title."""
+    logger.info('Searching for existing ArcGIS group "{}"'.format(groupTitle))
+    with util.CaptureStdoutLines() as output:
+        try:
+            arcGISAdmin = arcrest.manageorg.Administration(securityHandler=arcGIS.securityhandler)
+            group = getArcGISGroupByTitle(arcGISAdmin, groupTitle)
+        except Exception as exception:
+            logger.exception('Exception while searching for ArcGIS group "{}": {}'.format(groupTitle, exception))
+    if output:
+        logger.info('Unexpected output while searching for ArcGIS group "{}": {}'.format(groupTitle, output))
+    return group
+
+# Take two lists and separate out those only in first list, those only in second list, and those in both.
+# Uses to sets to do this so duplicate entries will become singular and list order will be arbitrary.
+def listDifferences(left_list, right_list):
```

Comment:
Underscores in symbol names annoys me (except for constants), so I avoided them in the rest of this program.  It would be nice to remove them from this method to be consistent.

Reply:
I'll standarize on camelCase.